Example/ReactiveDataDisplayManager/Table/DiffableTableViewController/DiffableTableViewController.swift

```diff
@@ -0,0 +1,145 @@
+//  ReactiveDataDisplayManagerExample
+//
+//  Created by Anton Eysner on 04.02.2021.
+//  Copyright © 2021 Alexander Kravchenkov. All rights reserved.
+//
+
+import UIKit
+import ReactiveDataDisplayManager
+
+@available(iOS 13.0, *)
+final class DiffableTableViewController: UIViewController {
+
+    // MARK: - Constants
+
+    private enum Constants {
+        static let titleForSection = "Section"
+        static let models = ["Afghanistan",
+                             "Afghanistan",
+                             "Albania",
+                             "Algeria",
+                             "Andorra",
+                             "Angola",
+                             "Antigua and Barbuda",
+                             "Argentina",
+                             "Armenia",
+                             "Australia",
+                             "Austria",
+                             "Azerbaijan",
+                             "Bahamas",
+                             "Bahrain",
+                             "Bangladesh",
+                             "Barbados",
+                             "Belarus"]
+    }
+
+    // MARK: - IBOutlets
+
+    @IBOutlet private weak var tableView: UITableView!
+
+    // MARK: - Private Properties
+
+    private lazy var adapter = tableView.rddm.manualBuilder
+        .set(dataSource: { DiffableTableDataSource(provider: $0) })
+        .build()
+
+    private var generators: [DiffableCellGenerator] = []
+
+    // MARK: - UIViewController
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = "Table with diffableDataSource"
+
+        setupSearch()
+        setupBarButtonItem()
+        fillAdapter()
+    }
+
+}
+
+// MARK: - Private methods
+
+@available(iOS 13.0, *)
+private extension DiffableTableViewController {
+
+    func setupSearch() {
+        let searchBar = UISearchBar()
+        searchBar.delegate = self
+        navigationItem.titleView = searchBar
+    }
+
+    func setupBarButtonItem() {
+        let button = UIBarButtonItem(title: "Remove First", style: .plain, target: self, action: #selector(removeFirst))
+        navigationItem.rightBarButtonItem = button
+    }
+
+    /// This method is used to fill adapter
+    func fillAdapter() {
+
+        generators = makeCellGenerators()
+
+        // add generators
+        adapter.addCellGenerators(generators)
+
+        // apply snapshot
+        adapter.forceRefill()
+    }
+
+    // Create cells generators
+    func makeCellGenerators() -> [DiffableCellGenerator] {
+         Constants.models.map { title in
+            DiffableCellGenerator(with: title)
+         }
+    }
+
+    // filter generators
+    func filterGenerators(with filter: String) -> [DiffableCellGenerator] {
+        guard !filter.isEmpty, !generators.isEmpty else {
+            return generators
+        }
+        return generators.filter { $0.model.starts(with: filter) }
+    }
+
+    @objc
+    func removeFirst() {
+
```

Comment:
Проверяем, что массив не пустой, чтобы избежать краш при удалении элемента из пустого массива

```suggestion
        guard !generators.isEmpty else { return }

```

Reply:
как ни странно крашей не было
но проверку я добавил)
может на какой-то из старых iOS и были краши

Reply:
Краш будет даже на iOS14
Swift, к сожалению, не делает проверку(( также как если попытаться получить элемент из массива по индексу, которого нет

https://user-images.githubusercontent.com/74650281/116509467-a4171f80-a8c3-11eb-9247-749569f484df.mov

Reply:
да
я ж проверил уже исправленный вариант
не проснулся еще видимо)